src/cljs/main/broadfcui/common/gcs_file_preview.cljs

```diff
@@ -128,7 +129,36 @@
                                  (when-not (string/blank? (@refs "preview"))
                                    (aset (@refs "preview") "scrollTop" (aget (@refs "preview") "scrollHeight"))))))})))})
 
-
+(react/defc DOSPreviewDialog
+  {:component-will-mount
+   (fn [{:keys [state props]}]
+     (let [{:keys [dos-uri]} props]
+       (swap! state assoc :translating-dos? true)
+       (ajax/call-martha (utils/->json-string {:url dos-uri, :pattern "gs://"})
+         {:on-done (fn [{:keys [success? raw-response xhr status-code get-parsed-response]}]
+                     (swap! state assoc
+                       :translating-dos? false
+                       :loading? true
```

Comment:
is `loading?` used anywhere?

Reply:
Yes in the preview modal above, "loading" and "translating" are two different things, both need to happen. After translating is done via Martha, loading starts from Google